tests/components/mystrom/test_config_flow.py

```diff
@@ -0,0 +1,129 @@
+"""Test the myStrom config flow."""
+from unittest.mock import AsyncMock, patch
+
+from pymystrom.exceptions import MyStromConnectionError
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.mystrom.const import DOMAIN
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+DEVICE_MAC = "6001940376EB"
+DEVICE_NAME = "myStrom Device"
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+async def test_form_combined(hass: HomeAssistant, mock_setup_entry: AsyncMock) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "pymystrom.get_device_info",
+        side_effect=AsyncMock(return_value={"type": 101, "mac": DEVICE_MAC}),
+    ) as mock_session:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "host": "1.1.1.1",
+            },
+        )
+        await hass.async_block_till_done()
+
+        assert result2["type"] == FlowResultType.CREATE_ENTRY
+        assert result2["title"] == DEVICE_NAME
+        assert result2["data"] == {"host": "1.1.1.1"}
+
+    # test for duplicates
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "pymystrom.get_device_info",
+        return_value={"type": 101, "mac": DEVICE_MAC},
+    ) as mock_session:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "host": "1.1.1.1",
+            },
+        )
+        await hass.async_block_till_done()
+
+        assert result2["type"] == FlowResultType.ABORT
+
+    assert len(mock_session.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_duplicates(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock
+) -> None:
+    """Test we get the form."""
```

Comment:
If you apply the changes above, I think this one can be removed.

Reply:
Removed